paper/paper.md

```diff
@@ -84,50 +110,99 @@ neighbors = grid.get_neighbors(pos, moore=True)
 ```python
 network = NetworkGrid(networkx_graph)
 network.get_neighbors(agent, include_center=False)
-```
+``` -->
+
+The OrthogonalMooreGrid, OrthogonalVonNeumanGrid and Hexgrid come with support for numpy based layers with additional data: PropertyLayers. Cells have attribute access to their value in each of these property layers, while the entire layer can be accessed from the space itself.
+
+<!-- ```python
+elevation = PropertyLayer("elevation", width, height)
+grid.add_property_layer(elevation)
+high_ground = grid.properties["elevation"].select_cells(lambda x: x > 50)
+``` -->
+
+
+
+
+2. Continuous spaces:
 
-3. Continuous spaces:
 ```python
 space = ContinuousSpace(x_max, y_max, torus=True)
 space.move_agent(agent, (new_x, new_y))
 ```
 
-Environmental properties can be modeled using PropertyLayers:
+
+
+
+### Time advancement
+Typically agent based models rely on discrete time advancement, or ticks. For each tick, the step method of the model is called. This in turn activates activates the agents in some way. The most frequent encountered approach is shown below, which runs a model for 100 ticks.
+
 ```python
-elevation = PropertyLayer("elevation", width, height)
-grid.add_property_layer(elevation)
-high_ground = grid.properties["elevation"].select_cells(lambda x: x > 50)
+
+model = Model(seed=42)
+
+for _ in range(100):
+    model.step()
+
 ```
 
-For more sophisticated environmental modeling, the experimental cell space system enables active cells with their own properties and behaviors:
+Generally, within the step method of the model, one activates all the agents. The AgentSet class can be used for this. Some common agent activation patterns are shown below
 
+1. Deterministic activation of agent
 ```python
-class ForestCell(Cell):
-    def step(self):
-        if self.on_fire:
-            self.spread_fire()
+model.agents.do("step")  # Random activation
 ```
 
-### Time management
-Mesa offers flexible approaches to time management:
-
-1. Simple step-based progression:
+2. Random activation of agents
 ```python
 model.agents.shuffle_do("step")  # Random activation
 ```
 
-2. Multi-stage activation:
+3. Multi-stage activation:
 ```python
 for stage in ["move", "eat", "reproduce"]:
     model.agents.do(stage)
 ```
 
-3. Event-based scheduling for non-uniform time steps:
+4. Activation by agent subclass:
+```python
+for klass in model.agent_types:
+    model.agents_by_type[klass].do("step")
+```
+
+Evidently, these activation patterns can be combined to create more sophisticated and complex activation patterns. For example
+
+4. Random activation of random agent subclass
 ```python
-scheduler = DiscreteEventScheduler(model)
-scheduler.add_event(SimEvent(time=3.5, target=agent))
+agent_klasses = model.agent_types
+model.random.shuffle(agent_klasses)
+for klass in agent_klasses:
+    model.agents_by_type[klass].shuffle_do("step")
 ```
```

Comment:
I think one of these is enough. Most of the time you want to control this explicitly anyway, like we do in Wolf Sheep:

```Python
        # First activate all sheep, then all wolves, both in random order
        self.agents_by_type[Sheep].shuffle_do("step")
        self.agents_by_type[Wolf].shuffle_do("step")
```